src/test/java/org/jenkinsci/plugins/durabletask/executors/OnceRetentionStrategyTest.java

```diff
@@ -0,0 +1,227 @@
+package org.jenkinsci.plugins.durabletask.executors;
+
+import java.io.File;
+import java.io.IOException;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.jenkinsci.plugins.durabletask.BourneShellScript;
+import org.jenkinsci.plugins.durabletask.WindowsBatchScript;
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.jvnet.hudson.test.BuildWatcher;
+import org.jvnet.hudson.test.Issue;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.LoggerRule;
+import org.jvnet.hudson.test.SimpleCommandLauncher;
+import hudson.Functions;
+import hudson.model.Computer;
+import hudson.model.LoadStatistics;
+import hudson.model.TaskListener;
+import hudson.model.Descriptor.FormException;
+import hudson.model.labels.LabelAtom;
+import hudson.remoting.RequestAbortedException;
+import hudson.remoting.Which;
+import hudson.slaves.AbstractCloudComputer;
+import hudson.slaves.AbstractCloudSlave;
+import hudson.slaves.Cloud;
+import hudson.slaves.NodeProvisioner.PlannedNode;
+import hudson.util.RemotingDiagnostics;
+import jenkins.model.Jenkins;
+
+import static org.awaitility.Awaitility.await;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.arrayWithSize;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.is;
+
+public class OnceRetentionStrategyTest {
+
+    private final static Logger LOGGER = Logger.getLogger(OnceRetentionStrategyTest.class.getName());
+
+    @Rule
+    public LoggerRule lr = new LoggerRule().record(LOGGER, Level.ALL);
+
+    @Rule
+    public JenkinsRule jr = new JenkinsRule();
+
+    @Rule
+    public TemporaryFolder temp = new TemporaryFolder();
+
+    @Rule
+    public BuildWatcher bw = new BuildWatcher();
+
+    @BeforeClass
+    public static void setupStatics() {
+        LoadStatistics.CLOCK = 100; // 100ms for the LoadStatistics so we have quick provisioning
+        // deadConnectionsShouldReLaunched seems to fail always when run as part of the suite - but passes always when run individually without the binary wrapper
+        // (because the ping command and its parent cmd.exe get killed :-o )
```

Comment:
this smells like a bug in the wrapper - however this is not something introduced here.
could be easy to reproduce by changing this to false for someone to investigate - but I beleive we wanted to change the default of the binary wrappers to be enabled anyway - so this seems like it could end up in being wasted effort.

Reply:
Yes, this has been a long-standing windows issue (see https://issues.jenkins.io/browse/JENKINS-47657) that was hopefully solved by the binary wrapper for windows. Using the script wrapper will definitely not work as the process will always be a child of the launching agent.
FWIW I have been running the majority of tests in powershellscript, powershellcore, etc, with the binary wrapper set as true.